server/public/model/cloud.go

```diff
@@ -333,3 +333,13 @@ func (df *Feedback) ToMap() map[string]any {
 
 	return res
 }
+
+type PatchInstallationRequest struct {
+	AllowedIPRanges  *AllowedIPRanges
```

Comment:
Normally I would, but the Provisioner's practices don't actually use snake case anywhere, they simply write the struct field as it is - so leaving this means I don't need to have multiple structs with a "translation layer" before passing data onwards

Reply:
I guess we don't often have objects making their way from the web app through to the provisioner. I'd rather have the web app code match its conventions as well, so I feel like we're going to need a translation layer of some sort to make that work.

I don't know if anyone has any strong feelings about the casing of API parameters, but I'd guess the easiest way to do that would be to define an anonymous struct type for these in one of the API layers, right before the object is encoded. We could do something like
```go
var req *model.PatchInstallationRequest

type RequestJSON struct {
    AllowedIPRanges *model.AllowedIPRanges `json:"allowed_ip_ranges"`
}
    
json.NewEncoder(w).Encode((*RequestJSON)(req))
```

Reply:
I'm not sure I follow - as is, it already works, because Go will just marshal the struct into JSON using the struct key (ie `AllowedIPRanges`). Having the types match that pascal casing makes everything line up. 

Or are you saying you'd prefer to have the webapp still conform to snake case?

Reply:
Yeah, I'd prefer to still send snake case to the web app and through the `api4` methods. I was never a huge fan of using snake case for it, but I'd rather stick to our standard behaviour than to end up with inconsistencies